application/cms/views.py

```diff
@@ -1176,6 +1190,71 @@ def create_data_source(topic_slug, subtopic_slug, measure_slug, version):
         )
 
 
+@cms_blueprint.route("/<topic_slug>/<subtopic_slug>/<measure_slug>/<version>/edit/data-sources/link", methods=["POST"])
+@login_required
+@user_has_access
+def link_existing_data_source(topic_slug, subtopic_slug, measure_slug, version):
+    topic, subtopic, measure, measure_version = page_service.get_measure_version_hierarchy(
+        topic_slug, subtopic_slug, measure_slug, version
+    )
+
+    form = SelectOrCreateDataSourceForm(data_sources=[])
+
+    if len(measure_version.data_sources) >= 2:
+        flash("Only two data sources can currently be linked to a measure version.", "error")
+
+        return redirect(
+            url_for(
+                "cms.edit_measure_version",
+                topic_slug=topic_slug,
+                subtopic_slug=subtopic_slug,
+                measure_slug=measure_slug,
+                version=measure_version.version,
+            )
+        )
+
+    if form.data_sources.data == CREATE_NEW_DATA_SOURCE:
+        return redirect(
+            url_for(
+                "cms.create_data_source",
+                topic_slug=topic_slug,
+                subtopic_slug=subtopic_slug,
+                measure_slug=measure_slug,
+                version=version,
+            )
+        )
+
+    elif form.data_sources.data:
+        data_source = DataSource.query.get(form.data_sources.data)
+
+        if data_source not in measure_version.data_sources:
```

Comment:
I think we should display an error (and redirect back to the search page) if the data source is already associated with the measure, rather than just silently ignore it and redirect back to the edit measure page?

Reply:
I originally had an error, but removed it because there isn't really anything the user can *do* about it.

But I'll add the error back. It's probably better to have something there at least, you're right.